core/src/test/java/org/elasticsearch/test/XContentTestUtils.java

```diff
@@ -46,58 +46,76 @@ private XContentTestUtils() {
 
 
     /**
-     * Compares to maps generated from XContentObjects. The order of elements in arrays is ignored
+     * Compares to maps generated from XContentObjects. The order of elements in arrays is ignored.
+     *
+     * @return null if maps are equal or path to the element where the difference was found
      */
-    public static boolean mapsEqualIgnoringArrayOrder(Map<String, Object> first, Map<String, Object> second) {
+    public static String mapsEqualIgnoringArrayOrder(String path, Map<String, Object> first, Map<String, Object> second) {
```

Comment:
This is kind of confusing, I think instead of trying to use this method as both a comparing method and an explaining method, maybe create an additional method called `differenceBetweenMaps` or something that returns the explanation and leave this one as a `boolean` method?

Otherwise the name doesn't really match its type, and it's confusing in the comparison to be using a string as the expected null value instead of the build-in junit explanation parameter on all the `assert*` methods.